vcr/credential/validator.go

```diff
@@ -78,28 +85,46 @@ func Validate(credential vc.VerifiableCredential) error {
 		return failure("'proof' is required")
 	}
 
-	return nil
+	return d.validateAllFieldsKnown(credential)
 }
 
-type defaultCredentialValidator struct{}
-
-func (d defaultCredentialValidator) Validate(credential vc.VerifiableCredential) error {
-	if err := Validate(credential); err != nil {
-		return err
+// validateAllFieldsKnown verifies that all fields in the VC are specified by the JSON-LD context.
+func (d defaultCredentialValidator) validateAllFieldsKnown(input vc.VerifiableCredential) error {
+	// First expand, then compact and marshal to JSON, then compare
+	inputAsJSON, _ := input.MarshalJSON()
+	inputAsMap := make(map[string]interface{})
+	_ = json.Unmarshal(inputAsJSON, &inputAsMap)
+
+	processor := ld.NewJsonLdProcessor()
+	options := ld.NewJsonLdOptions("") // TODO: why?
```

Comment:
doesn't matter for your own comparison. The given string acts as "root" context name.